etc/infrastructure/php/Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM php:7.2.10-fpm-alpine
+
+LABEL mantainer="sdecandelario@gmail.com"
+
+COPY --from=composer:1.7.2 /usr/bin/composer /usr/bin/composer
+
+RUN apk --update upgrade \
+    && apk add autoconf automake make gcc g++
+
+RUN apk add rabbitmq-c rabbitmq-c-dev \
+    && pecl install amqp apcu\
+    && docker-php-ext-enable amqp apcu
+
+CMD ["sh", "-c", "composer install && php-fpm"]
```

Comment:
Why composer install is container start? The provisioning should be in an early stage IMHO

Reply:
But I think for development is nice when you start the environment that install dependencies, thinking about someone new to the project. Or it's better to put in the `Dockerfile` instead?

Reply:
I would also go for the approach suggested by @jorge07. The reason why is because provisioning (running the `composer install`) in an early stage of our `Dockerfile` would allow us to perform some optimizations 😬

One of these optimizations would be applying the _builder pattern_ or _Multistage builds_. You can read more information about them in [the "Docker: De 0 a deploy" CodelyTV Pro course](https://pro.codely.tv/library/docker-de-0-a-deployment/70144/about/) by @fiunchinho (here you have [the course repository with some examples](https://github.com/fiunchinho/codely-docker/blob/master/building/multi-stage/Dockerfile_best#L6)), and [a blog post about this technic and its benefits](https://medium.com/@kaperys/create-lean-docker-images-using-the-builder-pattern-37fe2b5d97d4).

However, we've tried to apply this strategy and we've miserably failed (🤣). The reason why we haven't been able to do so is because the PHP `Dockerfile` is in a subdirectory and it can't copy the `composer.json` file (it is inside the parent folder).

I've opened an isolated issue in order to deal with this. This way we can go ahead merging all the improvements contained in this PR 🙂